### PR TITLE
Make all dictionary entries lowercase

### PR DESCRIPTION
**Pull request by bl-ue**
This PR is a proposal to, for consistency, make all dictionary entries (typo, suggestions, and hints) lowercase.

I'm doing this for two reasons:
1. because I've observed that proper capitalization of various proper nouns is hard enforce everywhere, and all over the dict there are messed up capitalizations.

2. There is **no** point to having them capitalized, because `build_dict()` converts them to lowercase before it uses them:
   https://github.com/codespell-project/codespell/blob/c6293a665b64d7d19e31b907a28335619a0b4923/codespell_lib/_codespell.py#L458-L461

### CODE REVIEW

**Comment by vikivivi** [comment]
Please do not remove capitalise suggestion been build in the dictionaries. See:
https://www.grammarly.com/blog/capitalization-countries-nationalities-languages/
https://www.scribendi.com/academy/articles/capitalization.en.html

In my opinion, the capitalise suggestion in dictionary.txt is supposed for codespell to suggest in capitalised manner in-according to the dictionary. But currently it does not seem to work as expected on my system to suggest in capitalised manner.

Expectation example:
```sh
$ echo amercia | codespell -
amercia ==> America
$ echo wedensday | codespell -
wedensday ==> Wednesday
```

**Comment by bl-ue** [comment]
Yeah, that's because of the lines I linked above — the capitalisation in the dictionaries serves _no_ purpose.

**Comment by bl-ue** [comment]
Ideally, we could make it support capitalisation as @larsoner's todo says.  I'll try it one of these days.

**Review by peternewman — CHANGES_REQUESTED** [review]
Straight reject, no changes possible.

From a technical perspective you've only done one of the dictionaries, unless there is no capitalisation in all the others, which seems unlikely.

"consistency" is not a sufficient reason to undo all the effort that has been built into the dictionary so far for this future feature. You also don't know if other people aren't using the codespell dictionaries in other tools and making use of this functionality.

**Comment by peternewman** [comment]
> Yeah, that's because of the lines I linked above — the capitalisation in the dictionaries serves _no_ purpose.

No, those 282 are the groundwork for the implementation of that TODO, there's no reason not to get the dictionary into shape before the code can use it. Yes the remainder of the 32,000 entries may need reviewing and capitalisation adding, but the existing content would make the functionality immediately useful and means it's just a coding problem, not a language one too.

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -18797,6 +18797,7 @@ mofification->modification
 mofified->modified
 mofifies->modifies
 mofify->modify
+mohammedan->muslim
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:18800]: You should add this in a separate PR though.

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -17423,7 +17423,7 @@ laod->load
 laoded->loaded
 laoding->loading
 laods->loads
-Laotion->Laotian
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:17426]: Suggest lotion too

**Review by peternewman — CHANGES_REQUESTED** [review]
I couldn't work out why the additions and removals didn't match...

**Comment by bl-ue** [comment]
😆 I thought it probably wasn't a very good idea, but I thought I'd get more feedback by opening a PR with some real changes than an issue.  I agree, we should fix the code to respect the capitalisation, which is something that I'm working on.

**Comment by bl-ue** [comment]
>I couldn't work out why the additions and removals didn't match...

Yeah, I meant for that line to go in another PR 🙂

Thank you for the input @peternewman — I'll add your suggestions in a new PR I'm going to open.

**Comment by peternewman** [comment]
> laughing I thought it probably wasn't a very good idea, but I thought I'd get more feedback by opening a PR with some real changes than an issue. I agree, we should fix the code to respect the capitalisation, which is something that I'm working on.

If it's integrated with the function that currently tries to guess the case of the suggestion should be half way there, but it might need an option to ignore the capitalisation anyway (or offer both suggestions in interactive mode if it's mixed case), we probably need to play around with a few scenarios.

**Pull request closed without merging**